doc/userdoc/contribute/nest_with_ides.rst

```diff
@@ -0,0 +1,103 @@
+Developing NEST with IDEs
+=========================
+
+Integrated development environments can make coding more efficient and fun.
+Here are some recipes based on practical experience on how to use
+IDEs in NEST development. The information is not meant to be complete,
+cross-checked or up-to-date but is provided on a best-effort basis to get
+you started. Kindly contribute your experiences.
+
+Since we generally recommend out-of-source builds, only such are discussed here.
+
+.. contents:: We currently have recipes for the following IDEs
+   :local:
+   :depth: 2
+
+Eclipse
+-------
+
+This recipe describes how to set up Eclipse for editing, building and
+running NEST. The description here was tested on macOS 11.
+
+Requirements and limitations
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+* Focus on single build configuration
+* Assumes all dependencies (OpenMPI, GSL, etc) installed in a Conda environment
+* Does not support debugging on macOS (because Eclipse does not support lldb)
+* Does not read the NEST `.clang-format` file, so code formatting may
+  be incorrect
+* Does not use the ``cmake4eclipse`` plug-in, because we haven't figure out
+  how to use it with our complex setup. The instructions below rely on running
+  ``cmake`` manually in the build directory.
+* Tested with Eclipse 2020-12
+
+Preparations
+~~~~~~~~~~~~
+
+#. Install Eclipse from `Eclipse Installer <https://www.eclipse.org/downloads/packages/installer>`_.
+   Using the Installer ensures that a suitable Java is installed. The instructions
+   below are based on choosing the *Eclipse IDE for Scientific Computing*.
+#. From the Eclipse Marketplace, install *PyDev* and *LiClipse Text* extensions.
+#. In Eclipse preferences, under ``General > Security > Secure Storage > Advanced``,
+   you should replace the default password encryption scheme to a more secure level
+   than default, e.g. to ``PBE...SHA512...AES_256``.
+
+Setting up the project
+~~~~~~~~~~~~~~~~~~~~~~
+
+#. :doc:`Clone NEST <development_workflow>` onto your computer
+#. Build NEST manually
+
+   a. Create a build directory
+   #. Run ``cmake`` with suitable options
+   #. Run ``make all`` and ``make install``
+#. In Eclipse
+
+   a. choose ``New > Makefile project with existing code``
+   #. select the directory containing the NEST source code
+   #. choose, e.g., the ``Cross GCC Toolchain``
+#. Right click the project and choose ``Properties`` from the context
+   menu
+
+   a. Under ``C/C++ Build/Build Variables``, define ``BUILD_DIR`` and ``CONDA_ENV``,
+      both of type ``Path``. The first should contain the full path to the build
+      directory you created above, the second the full path to your conda 
+      environment, usually something like ``.../miniconda3/envs/nest-dev``.
+   #. Under ``C/C++ Build > Builder Settings``,
+
+      #. uncheck ``Use default build command``
```

Comment:
I do not seem to find this, so it might be different for the Linux version? Also, I do not seem to have a `C/C++ Build -> Builder Settings` but I have a `C/C++ Build -> Settings`, so I assume it is the same. I am also using Eclipse 2020-12
This is how it looks for me:
![image](https://user-images.githubusercontent.com/21308850/109657895-4934a680-7b66-11eb-9c94-027076ecebe0.png)

Reply:
Made difference between submenu entries and tabs explicit.